sympy/solvers/solveset.py

```diff
@@ -1117,3 +1120,514 @@ def linsolve(system, *symbols):
     # Return solutions
     solution = FiniteSet(tuple(solution))
     return solution
+
+
+##############################################################################
+# ------------------------------nonlinsolve ---------------------------------#
+##############################################################################
+
+
+def substitution(system, symbols, result, known_symbols, exclude, all_symbols):
+    r"""
+    Solves the `system` using substitution method.
+
+    Parameters
+    ==========
+
+    system : list of equations.
+
+    symbols : list of unsolved symbols.
+
+    known_symbols : list of solved symbols.
+
+    result : If non of the symbols are solved then it is empty list, otherwise
+    list of already solved_syms symbols( dict symbol : value )
+
+    exclude : Set of expression which should not be zero with result.
+    Mostly denominator expression.
+
+    all_symbols : known_symbols + unsolved symbols.
+
+    Returns
+    =======
+    Returns a FiniteSet of ordered tuple.
+
+    Examples
+    ========
+
+    >>> from sympy.core.symbol import symbols
+    >>> x, y = symbols('x, y', real = True)
+    >>> from sympy.solvers.solveset import substitution
+    >>> substitution([x +y], [x], [{y : 1}], [y], set([]), [x, y])
+    {(-1, 1)}
+    >>> substitution([x +y], [x], [{y : 1}], [y], set([x + 1]), [y, x])
+    EmptySet()
+    >>> substitution([x +y], [x], [{y : 1}], [y], set([x - 1]), [y, x])
+    {(1, -1)}
+
+    """
+
+    from sympy.core.compatibility import ordered, default_sort_key
+    from sympy import Complement
+    from sympy.core.containers import Dict
+
+    def _ok_syms(e, sort=False):
```

Comment:
Use proper variable names, don't use names like `e`. Explicit is better than implicit.